contracts/finance/VestingConfidential.sol

```diff
@@ -0,0 +1,167 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.24;
+
+import {Clones} from "@openzeppelin/contracts/proxy/Clones.sol";
+import {FHE, externalEuint64, ebool, euint64} from "@fhevm/solidity/lib/FHE.sol";
+
+import {TFHESafeMath} from "../utils/TFHESafeMath.sol";
+import {ManagedVault} from "./ManagedVault.sol";
+import {ConfidentialFungibleToken} from "../token/ConfidentialFungibleToken.sol";
+
+contract VestingConfidential {
```

Comment:
Do we have any ERC we could rely on to avoid opinionated design choices?

Reply:
I don't believe so. IMO this isn't an area that an ERC would make sense.